### PR TITLE
HDDS-1793. Acceptance test of ozone-topology cluster is failing

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

1. Remove explicit container names from `ozone-topology`, since they are not handled by `execute_robot_test`
   (`must specify at least one container source`)
2. Allow more than 3 datanodes (`ozone-topology` has 4) in `wait_for_datanodes`

https://issues.apache.org/jira/browse/HDDS-1793

## How was this patch tested?

```
$ cd hadoop-ozone/dist/target/ozone-0.5.0-SNAPSHOT/compose/ozone-topology
$ ./test.sh
Removing network ozone-topology_net
WARNING: Network ozone-topology_net not found.
Creating network "ozone-topology_net" with driver "bridge"
Creating ozone-topology_om_1         ... done
Creating ozone-topology_scm_1        ... done
Creating ozone-topology_datanode_2_1 ... done
Creating ozone-topology_datanode_1_1 ... done
Creating ozone-topology_datanode_3_1 ... done
Creating ozone-topology_datanode_4_1 ... done
0 datanode is up and healthy (until now)
0 datanode is up and healthy (until now)
4 datanodes are up and registered to the scm
==============================================================================
ozone-topology-auditparser
==============================================================================
ozone-topology-auditparser.Auditparser :: Smoketest ozone cluster startup
==============================================================================
Initiating freon to generate data                                     | PASS |
------------------------------------------------------------------------------
Testing audit parser                                                  | PASS |
------------------------------------------------------------------------------
ozone-topology-auditparser.Auditparser :: Smoketest ozone cluster ... | PASS |
2 critical tests, 2 passed, 0 failed
2 tests total, 2 passed, 0 failed
==============================================================================
ozone-topology-auditparser                                            | PASS |
2 critical tests, 2 passed, 0 failed
2 tests total, 2 passed, 0 failed
==============================================================================
Output:  /tmp/smoketest/ozone-topology/result/robot-ozone-topology-ozone-topology-auditparser-om.xml
==============================================================================
ozone-topology-basic :: Smoketest ozone cluster startup
==============================================================================
Check webui static resources                                          | PASS |
------------------------------------------------------------------------------
Start freon testing                                                   | PASS |
------------------------------------------------------------------------------
ozone-topology-basic :: Smoketest ozone cluster startup               | PASS |
2 critical tests, 2 passed, 0 failed
2 tests total, 2 passed, 0 failed
==============================================================================
Output:  /tmp/smoketest/ozone-topology/result/robot-ozone-topology-ozone-topology-basic-scm.xml
Stopping ozone-topology_datanode_1_1 ... done
Stopping ozone-topology_datanode_2_1 ... done
Stopping ozone-topology_datanode_3_1 ... done
Stopping ozone-topology_datanode_4_1 ... done
Stopping ozone-topology_scm_1        ... done
Stopping ozone-topology_om_1         ... done
Removing ozone-topology_datanode_1_1 ... done
Removing ozone-topology_datanode_2_1 ... done
Removing ozone-topology_datanode_3_1 ... done
Removing ozone-topology_datanode_4_1 ... done
Removing ozone-topology_scm_1        ... done
Removing ozone-topology_om_1         ... done
Removing network ozone-topology_net
Log:     hadoop-ozone/dist/target/ozone-0.5.0-SNAPSHOT/compose/ozone-topology/result/log.html
Report:  hadoop-ozone/dist/target/ozone-0.5.0-SNAPSHOT/compose/ozone-topology/result/report.html
```

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 88 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | yamllint | 0 | yamllint was not available. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 614 | trunk passed |
| +1 | compile | 282 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 796 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 174 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 483 | the patch passed |
| +1 | compile | 286 | the patch passed |
| +1 | javac | 286 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 3 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 667 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 165 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 375 | hadoop-hdds in the patch passed. |
| -1 | unit | 2008 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 49 | The patch does not generate ASF License warnings. |
| | | 6208 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.TestMiniOzoneCluster |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestReadRetries |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1096 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient yamllint shellcheck shelldocs |
| uname | Linux 95aeb8bcda6f 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 5446308 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/1/testReport/ |
| Max. process+thread count | 4193 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/dist U: hadoop-ozone/dist |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by adoroszlai** [comment]
(force-pushed to pick up the fix for `author.sh` from `trunk`)

File: hadoop-ozone/dist/src/main/compose/ozone-topology/test.sh [hunk 1]
```diff
@@ -15,6 +15,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+DATANODE_COUNT=4
```
Comment (hadoop-yetus) [inline hadoop-ozone/dist/src/main/compose/ozone-topology/test.sh]: shellcheck:1: warning: DATANODE_COUNT appears unused. Verify it or export it. [SC2034]

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 109 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | yamllint | 0 | yamllint was not available. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 527 | trunk passed |
| +1 | compile | 271 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 808 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 169 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 469 | the patch passed |
| +1 | compile | 277 | the patch passed |
| +1 | javac | 277 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | shellcheck | 0 | The patch generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 683 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 164 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 457 | hadoop-hdds in the patch passed. |
| -1 | unit | 2589 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 6787 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.scm.TestXceiverClientManager |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.TestMiniOzoneCluster |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerByPipeline |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1096 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient yamllint shellcheck shelldocs |
| uname | Linux ca7225d9159d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1411513 |
| Default Java | 1.8.0_212 |
| shellcheck | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/3/artifact/out/diff-patch-shellcheck.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/3/testReport/ |
| Max. process+thread count | 4320 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/dist U: hadoop-ozone/dist |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/dist/src/main/compose/testlib.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ mkdir -p "$RESULT_DIR"
 #Should be writeable from the docker containers where user is different.
 chmod ogu+w "$RESULT_DIR"
 
-## @description wait until 3 datanodes are up (or 30 seconds)
+## @description wait until 3 or more datanodes are up (or 30 seconds)
 ## @param the docker-compose file
 wait_for_datanodes(){
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/dist/src/main/compose/testlib.sh]: Thanks for the suggestion.  I have replaced the global variable with a function parameter in a new commit.  This also gets rid of the shellcheck warning.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| 0 | yamllint | 0 | yamllint was not available. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 495 | trunk passed |
| +1 | compile | 294 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 849 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 175 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 453 | the patch passed |
| +1 | compile | 286 | the patch passed |
| +1 | javac | 286 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | There were no new shellcheck issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 680 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 176 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 314 | hadoop-hdds in the patch passed. |
| -1 | unit | 2492 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 76 | The patch does not generate ASF License warnings. |
| | | 6571 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.web.client.TestBuckets |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.web.client.TestKeys |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1096 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient yamllint shellcheck shelldocs |
| uname | Linux 22e2f3409154 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ef66e49 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/4/testReport/ |
| Max. process+thread count | 5240 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/dist U: hadoop-ozone/dist |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1096/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by elek** [comment]
(It's failing because same space issue on the jenkins. I am cleaning it up...)

**Comment by elek** [comment]
yeah, full green acceptance test. I am committing it to trunk/ozone-0.4.1...

**Pull request closed without merging**